src/game/utils/batch/report.rs

```diff
@@ -0,0 +1,296 @@
+// I would rather not have to replay histories here b/c it is complicated and possibly slow
+// NOTE: Since this mutates the game, it un-caches weights
+fn get_action_evs(game: &mut PostFlopGame) -> Vec<f32> {
+    let actions = game.available_actions();
+    (0..actions.len())
+        .map(|action_index| {
+            let player = game.current_player();
+            let history = game.history().to_owned();
+            game.play(action_index);
+            game.cache_normalized_weights();
+
+            let evs = game.expected_values(player);
+            let average_ev = evs.iter().sum::<f32>() / evs.len() as f32;
+
+            game.back_to_root();
+            game.apply_history(history.as_slice());
+
+            average_ev
+        })
+        .collect()
+}
+
+fn is_terminating_action(action: Action, player: usize) -> bool {
+    match action {
+        Action::AllIn(_) => false,
+        Action::Bet(_) => false,
+        Action::Raise(_) => false,
+        // Check is terminating iff current player is IP
+        Action::Check => player == 1,
+        _ => true,
+    }
+}
+
+fn folder_name_from_action(action: Action) -> String {
+    match action {
+        Action::AllIn(x) => format!("allin{}", x),
+        Action::Bet(x) => format!("bet{}", x),
+        Action::Raise(x) => format!("raise{}", x),
+        Action::Check => "check".to_string(),
+        _ => unimplemented!("Cannot currently make folder for terminating action"),
+    }
+}
+
+fn fmt_floats(floats: &[f32], formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    write!(
+        formatter,
+        "{}",
+        floats
+            .iter()
+            .map(|x| format!("{:.2}", x))
+            .collect::<Vec<String>>()
+            .join(",")
+    )
+}
+
+fn report_header(actions: &Vec<Action>) -> String {
+    format!(
+        "Flop,IP Eq,IP EV,IP EQR,OOP Eq,OOP EV,OOP EQR,{}",
+        // Title for likelihood & EV per action
+        actions
+            .iter()
+            .map(|a| format!("{:?},{:?} EV", a, a))
+            .collect::<Vec<String>>()
+            .join(","),
+    )
+}
+
+/// Single row in an aggregate report
+/// flop -- describes the 3 cards of the flop
+/// ip_equity -- equity (0-100 value) of the in-position player
+/// ip_ev -- expected value (in chips) of the in-position player
+/// ip_eqr -- equity realization (0-100 value) of the in-position player
+/// oop_equity -- equity (0-100 value) of the out-of-position player
+/// oop_ev -- expected value (in chips) of the out-of-position player
+/// oop_eqr -- equity realization (0-100 value) of the out-of-position player
+/// actions -- list of action likelihoods (0-100), corresponding to the list of action from the owning AggActionTree
+/// action_evs -- expected value (in chips) resulting from each action
+pub struct AggRow {
+    flop: [u8; 3],
+    ip_equity: f32,
+    ip_ev: f32,
+    ip_eqr: f32,
+    oop_equity: f32,
+    oop_ev: f32,
+    oop_eqr: f32,
+    actions: Vec<f32>,
```

Comment:
I think we should call this `action_frequencies` or `action_freqs` explicitly just for clarity